analyses/sample-distribution-analysis/01-filter-across-types.R

```diff
@@ -0,0 +1,225 @@
+# This script filters the given dataset to produce a summarized visualization
+# of key variables within the dataset.
+#
+# Chante Bethell for CCDL 2019
+
+
+# Load/install packages. This will be removed once a Dockerfile is established.
+source("00-install-packages.R")
+
+# magrittr pipe
+`%>%` <- dplyr::`%>%`
+
+# Function to filter dataset based on primary_site of disease, while arranging
+# the cancer types in order of descending expression.
+location_fn <- function(location) {
+  brain_location %>%
+    dplyr::arrange(desc(n)) %>%
+    dplyr::filter(stringr::str_detect(primary_site, location))
+}
+
+# Function to pad x with NA's
+na_pad <- function(x, len) {
+  x[1:len]
+}
+
+# Function to make a padded data.frame
+make_padded_data_frame <- function(l, ...) {
+  maxlen <- max(sapply(l, length))
+  data.frame(lapply(l, na_pad, len = maxlen), ...)
+}
+
+# Create directories to hold the output.
+if (!dir.exists("results")) {
+  dir.create("results")
+}
+if (!dir.exists("plots")) {
+  dir.create("plots")
+}
+
+# Read in dataset
+df2 <- data.frame(readr::read_tsv(
+  file.path("..", "..", "data","pbta-histologies.tsv")
+))
+
+# Remove na's
+df2 <- df2 %>%
+  dplyr::filter(!is.na(disease_type_new))
+
+# data.frame with the count of each unique cancer type expression
+disease_expression <- df2 %>%
+  dplyr::group_by(disease_type_new) %>%
+  dplyr::filter(!is.na(disease_type_new)) %>%
+  dplyr::count() %>%
+  dplyr::arrange(desc(n)) %>%
+  dplyr::rename(count = n)
+
+# Calculate the total count of the dataset
+sum_count <- sum(disease_expression$count)
+
+# Create a percent variable
+disease_expression <- disease_expression %>%
+  dplyr::mutate(percent = (count / sum_count))
```

Comment:
Out of the above suggestions, this is what worked: `dplyr::mutate(percent = paste0(((count / sum_count) * 100)`.
Would this still be a better solution compared to the inclusion of lines 65-66?

Reply:
Ah yes good call. Wouldn't you still need to add the `"%"` or am I misunderstanding the other part of what `formattable::percent` does?

Reply:
Ah yes, good catch. I have the `"%"` in the script but left it out when I copied the line here.

Reply:
I would say since we will have a monolithic Docker container for this project, any place where we can get rid of dependencies we should try and do that. This seems like something where we don't need a well-used function from a package to do it.

Reply:
Understood. I'll remove the use of that package and keep this in mind before implementing any other new packages in other scripts.